tracer/src/Datadog.Trace/SpanContext.cs

```diff
@@ -447,6 +447,17 @@ internal void SetCheckpoint(DataStreamsManager manager, CheckpointKind checkpoin
             }
         }
 
+        /// <summary>
+        /// There shouldn't be any need to manually set the pathway context to a known value,
+        /// except in the case where messages are consumed in batch, and then processed individually to produce more messages,
+        /// in which case we need to recover the consume checkpoint so that the produce checkpoint is properly linked to it.
+        /// Kafka is the only integration offering that feature for now.
+        /// </summary>
+        internal void ManuallySetPathwayContextToPairMessages(PathwayContext? pathwayContext)
+        {
+            PathwayContext = pathwayContext;
```

Comment:
Should we be passing in `DataStreamsManager manager` here and checking for `manager != null` 🤔 It's certainly not _required_ but I wonder if there's any logic somewhere that assumes that _if_ `PathwayContext != null` then `DataStreamsManager.Instance !=null` which isn't necessarily true here.

Alternatively, we _could_ add that check at the caller.

Reply:
We already have that check at the call site, we only extract and set `if(Tracer.Instance.TracerManager.DataStreamsManager is { IsEnabled: true })`. I can add this here as well out of extra caution, but that'd be really heavy in my opinion.

Reply:
> I can add this here as well out of extra caution, but that'd be really heavy in my opinion.

Meh, I wouldn't do it in both places, I only suggested moving the check in here to make sure we don't forget when we call it elsewhere too, but meh